### DISCUSSION TITLE
Prescribing realistic bottom bathymetry

### DISCUSSION THREAD

**Discussion by roxyboy** [CliMA/Oceananigans.jl#3390] [Computational science!]
Hi Oceananigans,

Thank you very much for making your code open!
@olmozavala and I are currently testing out to see if we can run realistic simulations of the Gulf of Mexico but the documentation (https://clima.github.io/OceananigansDocumentation/stable/appendix/library/#Oceananigans.Grids.LatitudeLongitudeGrid) seems to only have examples with a flat bottom.

Could you advise how one would prescribe a realistic ocean bathymetry?

Thank you.

**Comment by navidcy** [comment]
Hi @roxyboy, we'll make some docs/examples soon! There are some tools in a different repo but they are rough on the edges at the moment and I'm bit reluctant to point you to it.

If you have the `bathymetry` loaded as an array of the same size as a flat-bottom latitude-longitude grid then you can use [GridFittedBottom](https://clima.github.io/OceananigansDocumentation/stable/appendix/library/#Oceananigans.ImmersedBoundaries.GridFittedBottom) to do something like:

```Julia
underlying_grid = LatitudeLongitudeGrid(arch,
                                        size = (Nx, Ny, Nz),
                                        longitude = (-180, 180),
                                        latitude = (-75, 75),
                                        z = (-depth, 0),
                                        topology = (Periodic, Bounded, Bounded))

grid = ImmersedBoundaryGrid(underlying_grid, GridFittedBottom(bathymetry))
```

**Comment by glwagner** [comment]
@navidcy is a tease! The "other repo" is https://github.com/CliMA/ClimaOcean.jl. That's where we plan to build out the features needed for high-quality realistic simulations, eg interpolating bathymetry from one grid to another, "recipes for realism" such as parameterization and boundary condition choices. There's already a tool there for downloading some repeat-year forcing JRA55 data; we still need to build a tool for the entire dataset, and also tools for restricting to some region, etc. The vision in the end though is to get as close as possible to "one click regional modeling" as we can.

@navidcy is right that the tools are not so useful now unless you want to get into the weeds. But if you want to help, it's greatly appreciated. Feedback on issues or wishlist features could also be fun to discuss.

**Comment by glwagner** [comment]
Here's my latest work on setting up an OMIP simulation:

https://github.com/CliMA/ClimaOcean.jl/blob/glw-ss/ice-ocean-model/experiments/prototype_omip_simulation/omip_simulation.jl

which also includes an example of using `ImmersedBoundaryGrid` to implement bathymetry.

Bathymetry is a little sad there; I "diagnose" it from the ECCO state rather than prescribing directly. We need initialization tools for reasonable "in painting" for cases where initial condition information and bathymetry are slightly different (ie filling in NaNs near boundaries, etc). All of this should evolve quite a bit in the next few months.

PS I'm going to convert this to a Discussion so that we can continue the party without needing to make any source code changes.

**Comment by glwagner** [comment]
Note, this is not a "realistic" example but @navidcy is putting together an idealized bathymetry example here:

https://github.com/CliMA/Oceananigans.jl/blob/ncc/tide/examples/internal_tide.jl

Which is on PR https://github.com/CliMA/Oceananigans.jl/pull/3132

Problem is we are so darn busy with OMIP and global stuff we barely have time to build up docs. It could be nice to have help from the people who stand to benefit!
